src/main/java/com/greenfox/lvlup/model/entity/BadgeLevel.java

```diff
@@ -0,0 +1,98 @@
+package com.greenfox.lvlup.model.entity;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.persistence.*;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Data
+@EqualsAndHashCode(exclude = "holders")
+@Entity
+@Table(name = "badgelevels")
+public class BadgeLevel {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+    private String name;
+    private int level;
+    private String description;
+
+    @ManyToOne
+    private Badge badge;
+
+    @ManyToMany
+    @JoinTable(name = "badgelevel_user",
+            joinColumns = @JoinColumn(name = "badgelevel_id", referencedColumnName = "id"),
+            inverseJoinColumns = @JoinColumn(name = "user_id", referencedColumnName = "id"))
+    private Set<User> holders;
+
+    public BadgeLevel() {
+    }
+
+    public BadgeLevel(String name, int level, String description, Badge badge, User... holders) {
+        this.name = name;
+        this.level = level;
+        this.description = description;
+        this.badge = badge;
+        this.holders = Stream.of(holders).collect(Collectors.toSet());
+        this.holders.forEach(x -> x.getBagdes().add(this));
```

Comment:
Suvi, could we please have a closer look at these last two lines of the method? Both Andi and I would like to be able to explain to ourselves what exactly they do, especially the latter one. Thanks :)

Reply:
Sure, just don't forget to ask me on our daily meeting! :)